flink-kubernetes-operator/src/main/java/org/apache/flink/kubernetes/operator/reconciler/deployment/AbstractFlinkResourceReconciler.java

```diff
@@ -186,7 +187,41 @@ private void applyAutoscaler(FlinkResourceContext<CR> ctx) throws Exception {
                 ctx.getResource().getSpec().getJob() != null
                         && ctx.getObserveConfig().getBoolean(AUTOSCALER_ENABLED);
         autoScalerCtx.getConfiguration().set(AUTOSCALER_ENABLED, autoscalerEnabled);
+
         autoscaler.scale(autoScalerCtx);
+        putBackOldParallelismOverridesIfNewOnesAreMerelyAPermutation(ctx);
+    }
+
+    private static <
+                    CR extends AbstractFlinkResource<SPEC, STATUS>,
+                    SPEC extends AbstractFlinkSpec,
+                    STATUS extends CommonStatus<SPEC>>
```

Comment:
Can we rename this to `resetParallelismOverridesIfUnchanged` ? The current naming is unusually verbose for the codebase. It may be better to add a javadoc comment instead. But this may be irrelevant if you check my other comment. We don't need to replace it if we don't set it in the first place, but for that we need to move the logic